plugins/auto-sizes/tests/test-auto-sizes.php

```diff
@@ -279,4 +288,101 @@ public function test_auto_sizes_update_content_img_tag( string $input, string $e
 			auto_sizes_update_content_img_tag( $input )
 		);
 	}
+
+	/**
+	 * @covers ::auto_sizes_attribute_includes_valid_auto
+	 */
+	public function test_auto_sizes_attribute_includes_valid_auto(): void {
+		// Test when 'auto' is the first item in the list.
+		$this->assertTrue( auto_sizes_attribute_includes_valid_auto( 'auto, 100vw' ) );
+
+		// Test when 'auto' is not the first item in the list.
+		$this->assertFalse( auto_sizes_attribute_includes_valid_auto( '100vw, auto' ) );
+	}
+
+	/**
+	 * Provides data for the auto_sizes_update_image_attributes test.
+	 *
+	 * @return array<int,mixed[]> The data for the test cases.
+	 */
+	public function data_provider_for_auto_sizes_update_image_attributes(): array {
+		return array(
+			array(
+				array(
+					'loading' => 'eager',
+					'sizes'   => '100vw',
+				),
+				array(
+					'loading' => 'eager',
+					'sizes'   => '100vw',
+				),
+			),
+
+			// Test when the image is not responsive.
+			array( array( 'loading' => 'lazy' ), array( 'loading' => 'lazy' ) ),
+
+			// Test when 'auto' already exists in the sizes attribute.
+			array(
+				array(
+					'loading' => 'lazy',
+					'sizes'   => 'auto, 100vw',
+				),
+				array(
+					'loading' => 'lazy',
+					'sizes'   => 'auto, 100vw',
+				),
+			),
+
+			// Test when 'auto' needs to be added to the sizes attribute.
+			array(
+				array(
+					'loading' => 'lazy',
+					'sizes'   => '100vw',
+				),
+				array(
+					'loading' => 'lazy',
+					'sizes'   => 'auto, 100vw',
+				),
+			),
```

Comment:
This would be better as associative arrays as then it becomes self-documenting, and the array keys of the top-level can be used in filtering.

For example for the first item it would be better as:

```php
			'eager_something_blah_blah' => array(
				'attr'     => array(
					'loading' => 'eager',
					'sizes'   => '100vw',
				),
				'expected' => array(
					'loading' => 'eager',
					'sizes'   => '100vw',
				),
			),
```

Naturally you can use the appropriate labels for the keys 😄

Reply:
Addressed in https://github.com/WordPress/performance/pull/1879/commits/eaa6dfa4a60355ada269a311076970765ae59d75